frontend/src/components/common/authorized-route/authorized-route.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Route, Redirect, RouteProps } from "react-router-dom";
+import { useSelector } from 'react-redux';
+import { RootState } from 'common/types';
+import { AppRoute, PermissionKey, PermissionName } from 'common/enums';
+import { IPermission } from "common/interfaces";
+
+interface  IRouteProps extends RouteProps {
+  redirectTo?: AppRoute,
+  permissions?: PermissionName[],
+  component: React.ComponentType<any>,
+}
+
+const checkPermission = (checkListPermissionName: PermissionName[], Permissions: IPermission[]) =>
```

Comment:
Too hard, try to simplify and move it to the `frontend/src/components/common/authorized-route/helpers`